_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,247 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that has the [semantic role](#semantic-role) of [rowheader](https://www.w3.org/TR/wai-aria-1.1/#rowheader) or [columnheader](https://www.w3.org/TR/wai-aria-1.1/#columnheader) and is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of an element that:
```

Comment:
By looking at descendent, you're making an assumption that the header is part of the same table. That's not necessarily the case. I think it's better to keep the relationship explicit. The header has to be a child of a row, and the row has to be a child of either a table|grid or a group that is a child of a table|grid. I know that's a little harder to document, but that's far more accurate, and it's likely how anyone who tests this will look at it.

You're probably going to want to add some definitions for this to avoid having to write this all out into the applicability.

Reply:
Agreed.
```html
<table id="ok">
<td>
  <table id="ko" style="display: hidden">
    <th>bla</th>
  </table>
</td>
</table>
```
The `th` is a descendant of `ok` which is matching the conditions of the applicability. But the `th` doesn't need to be checked because it's actually a header in `ko` which doesn't match the conditions.

Another way to express it could be "whose closest table/grid ancestor is …", I think.